api_testrail.py

```diff
@@ -250,6 +257,67 @@ def testrail_milestones(self, project):
                 else:
                     print(f"No milestones data to insert into database for project {testrail_project_id}.") # noqa
 
+    def testrail_users(self):
+        # Step 1: Get all projects
+        projects_response = self.projects()
+        all_projects = projects_response.get("projects", [])
+        all_users = []  # List of all users across all projects
+        seen_project_ids = set()
+        project_user_counts = {}
+
+        for project in all_projects:
+            project_id = project["id"]
+            project_name = project["name"]
+
+            # Avoid hitting duplicate project IDs if that ever happens
+            if project_id in seen_project_ids:
+                continue
+            seen_project_ids.add(project_id)
+
+            try:
+                user_response = self.users(project_id)
+                users = user_response.get("users", [])
+                all_users.extend(users)
+
+                unique_emails = {user.get("email") for user in users if user.get("email")} # noqa
+                project_user_counts[project_name] = len(unique_emails)
+
+                print(f"{project_name} (ID: {project_id}): {len(unique_emails)} unique users (by email)") # noqa
+
+            except Exception as e:
+                print(f"Error fetching users {project_id} ({project_name}): {e}") # noqa
+
+        # Deduplicate all users globally by email
+        unique_by_email = {}
+        for user in all_users:
+            email = user.get("email")
+            if email:
+                unique_by_email[email] = user
+
+        print(f"\n Total unique users across all accessible projects (by email): {len(unique_by_email)}") # noqa
+
+        # Optional: pretty-print list
```

Comment:
not: Its not so clear how this is optional.  If this is diagnostic, perhaps better to say "# diagnostic" or something similar.